Reports the number of times the best tree score is reached.

In [1]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt
import seaborn as sns;
from collections import Counter
import glob
from tqdm import tqdm_notebook as tqdm
import os 
%matplotlib inline
sns.set(rc={'figure.figsize':(9.7,4.27)})

In [3]:
output_path = "/cluster/work/bewi/members/tuncel/sc-dna/notebooks/manuscript_simulations_max_tree_reproducibility/"
all_paths = glob.glob("/cluster/work/bewi/members/tuncel/data/dna/manuscript_/iterative_simulations/*")
scores_df = pd.DataFrame()
for path in tqdm(sorted(all_paths)): # 27
    #import ipdb; ipdb.set_trace()
    dups_per_sim = []
    for sim_id in range(0,100): # 100
        trees_in_folder_in_sim = glob.glob(path+'/'+str(sim_id)+'_*_tree_inferred.txt')
        tree_scores = [] 
        for tree_file in trees_in_folder_in_sim: # 10
            tree_score = np.genfromtxt(tree_file)[0][2]
            tree_scores.append(tree_score)
        max_ts = max(tree_scores)
        max_tree_scores = [ts for ts in tree_scores if ts==max_ts]
        n_dups = len(max_tree_scores) - len(set(max_tree_scores))
        dups_per_sim.append(n_dups)

    scores_df[path.split('/')[-1]] = dups_per_sim
scores_df.to_csv(output_path+"determining_n_iters_8192_coeff" + '.csv')

A Jupyter Widget

In [4]:
pd.set_option('display.max_columns', None)  # or 1000
pd.set_option('display.max_rows', None)  # or 1000
pd.set_option('display.max_colwidth', -1)  # or 199

In [5]:
scores_df.shape

(100, 27)

In [6]:
scores_df

10nodes_10regions_100000reads  10nodes_10regions_10000reads  \
0   8                              8                              
1   9                              7                              
2   7                              7                              
3   6                              9                              
4   9                              6                              
5   9                              8                              
6   7                              7                              
7   8                              6                              
8   6                              9                              
9   9                              6                              
10  8                              9                              
11  8                              6                              
12  8                              9                              
13  7                              8                              
14  8                              9                              
15  8                              9                              
16  8                              9                              
17  9                              7                              
18  7                              7                              
19  9                              8                              
20  8                              9                              
21  7                              7                              
22  9                              8                              
23  8                              9                              
24  6                              8                              
25  8                              8                              
26  8                              7                              
27  6                              9                              
28  8                              9                              
29  6                              9                              
30  6                              9                              
31  8                              7                              
32  9                              7                              
33  8                              6                              
34  9                              9                              
35  7                              9                              
36  8                              7                              
37  7                              8                              
38  8                              9                              
39  7                              7                              
40  8                              6                              
41  9                              6                              
42  6                              7                              
43  6                              7                              
44  9                              6                              
45  8                              9                              
46  8                              7                              
47  8                              9                              
48  9                              9                              
49  6                              9                              
50  7                              8                              
51  7                              8                              
52  7                              8                              
53  9                              8                              
54  9                              8                              
55  7                              7                              
56  9                              7                              
57  7                              7                              
58  6                              6               

In [7]:
scores_df.min(axis=0)

10nodes_10regions_100000reads     6
10nodes_10regions_10000reads      6
10nodes_10regions_30000reads      6
10nodes_20regions_100000reads     6
10nodes_20regions_10000reads      6
10nodes_20regions_30000reads      6
10nodes_40regions_100000reads     6
10nodes_40regions_10000reads      6
10nodes_40regions_30000reads      6
20nodes_20regions_100000reads     8
20nodes_20regions_10000reads      7
20nodes_20regions_30000reads      8
20nodes_40regions_100000reads     6
20nodes_40regions_10000reads      5
20nodes_40regions_30000reads      6
20nodes_80regions_100000reads     7
20nodes_80regions_10000reads      5
20nodes_80regions_30000reads      5
30nodes_120regions_100000reads    6
30nodes_120regions_10000reads     0
30nodes_120regions_30000reads     5
30nodes_30regions_100000reads     7
30nodes_30regions_10000reads      6
30nodes_30regions_30000reads      7
30nodes_60regions_100000reads     6
30nodes_60regions_10000reads      6
30nodes_60regions_30000reads      6
dtype: int64

In [8]:
scores_df.mean()

10nodes_10regions_100000reads     7.67
10nodes_10regions_10000reads      7.73
10nodes_10regions_30000reads      7.63
10nodes_20regions_100000reads     7.84
10nodes_20regions_10000reads      8.05
10nodes_20regions_30000reads      8.11
10nodes_40regions_100000reads     8.36
10nodes_40regions_10000reads      8.30
10nodes_40regions_30000reads      8.41
20nodes_20regions_100000reads     8.89
20nodes_20regions_10000reads      8.90
20nodes_20regions_30000reads      8.90
20nodes_40regions_100000reads     8.17
20nodes_40regions_10000reads      8.31
20nodes_40regions_30000reads      8.22
20nodes_80regions_100000reads     8.62
20nodes_80regions_10000reads      8.35
20nodes_80regions_30000reads      8.32
30nodes_120regions_100000reads    8.23
30nodes_120regions_10000reads     7.16
30nodes_120regions_30000reads     8.41
30nodes_30regions_100000reads     8.91
30nodes_30regions_10000reads      8.82
30nodes_30regions_30000reads      8.92
30nodes_60regions_100000reads     8.18
30nodes_60regions_10000re

In [9]:
42

42

### Renaming the failed ones

In [11]:
scores_df_cols = scores_df.columns.values

In [14]:
for col in scores_df_cols:
    file_path = "/cluster/work/bewi/members/tuncel/data/dna/manuscript_/iterative_simulations/"+col+'/'
    values_to_replace = scores_df.index[scores_df[col]<5].values
    print(col)
    print(values_to_replace)
    for value in values_to_replace:
        for filename in os.listdir(file_path):
            if filename.startswith(str(value)+"_"):
                print(filename)
                new_name = ".rep6_"+filename
                # os.rename(file_path+filename, file_path+new_name)

10nodes_10regions_100000reads
[]
10nodes_10regions_10000reads
[]
10nodes_10regions_30000reads
[]
10nodes_20regions_100000reads
[]
10nodes_20regions_10000reads
[]
10nodes_20regions_30000reads
[]
10nodes_40regions_100000reads
[]
10nodes_40regions_10000reads
[]
10nodes_40regions_30000reads
[]
20nodes_20regions_100000reads
[]
20nodes_20regions_10000reads
[]
20nodes_20regions_30000reads
[]
20nodes_40regions_100000reads
[]
20nodes_40regions_10000reads
[]
20nodes_40regions_30000reads
[]
20nodes_80regions_100000reads
[]
20nodes_80regions_10000reads
[]
20nodes_80regions_30000reads
[]
30nodes_120regions_100000reads
[]
30nodes_120regions_10000reads
[ 2 58 63 90]
2_infrep2_inferred_cnvs.txt
2_infrep7_inferred_cnvs.txt
2_infrep9_inferred_cnvs.txt
2_infrep5_tree_inferred.txt
2_infrep0_inferred_cnvs.txt
2_infrep1_tree_inferred.txt
2_infrep8_inferred_cnvs.txt
2_infrep7_tree_inferred.txt
2_infrep2_tree_inferred.txt
2_infrep8_tree_inferred.txt
2_infrep9_tree_inferred.txt
2_infrep5_inferred_cnvs.txt
2_in

In [27]:
scores_df.index[scores_df["10nodes_20regions_100000reads"]<=5]

Int64Index([], dtype='int64')

In [23]:
scores_df

10nodes_10regions_100000reads  10nodes_10regions_10000reads  \
0   8                              8                              
1   9                              7                              
2   7                              7                              
3   6                              9                              
4   9                              6                              
5   9                              8                              
6   7                              5                              
7   8                              6                              
8   6                              9                              
9   9                              6                              
10  8                              9                              
11  8                              6                              
12  8                              9                              
13  7                              8                              
14  8                              9                              
15  8                              9                              
16  8                              9                              
17  9                              7                              
18  7                              7                              
19  9                              8                              
20  8                              9                              
21  7                              7                              
22  9                              8                              
23  8                              9                              
24  6                              8                              
25  8                              8                              
26  8                              7                              
27  6                              9                              
28  8                              9                              
29  6                              9                              
30  6                              9                              
31  8                              7                              
32  9                              7                              
33  8                              6                              
34  9                              9                              
35  7                              9                              
36  8                              7                              
37  7                              8                              
38  8                              9                              
39  7                              7                              
40  8                              6                              
41  9                              6                              
42  6                              7                              
43  6                              7                              
44  9                              6                              
45  8                              9                              
46  8                              7                              
47  8                              9                              
48  9                              9                              
49  6                              9                              
50  7                              8                              
51  7                              8                              
52  7                              8                              
53  9                              8                              
54  9                              8                              
55  7                              7                              
56  9                              7                              
57  7                              7                              
58  6                              5               

In [ ]:
scores_df.head()

In [ ]:
5

In [ ]:
5

In [16]:
scores_df.mean()

10nodes_10regions_100000reads         7.37
10nodes_10regions_10000reads          7.26
10nodes_10regions_30000reads          7.48
10nodes_20regions_100000reads         7.80
10nodes_20regions_10000reads          7.85
10nodes_20regions_30000reads          7.94
10nodes_40regions_100000reads         8.36
10nodes_40regions_10000reads          8.02
10nodes_40regions_30000reads          8.27
20nodes_20regions_100000reads         8.89
20nodes_20regions_10000reads          8.90
20nodes_20regions_30000reads          8.90
20nodes_40regions_100000reads         8.13
20nodes_40regions_10000reads          8.07
20nodes_40regions_30000reads          8.14
20nodes_80regions_100000reads         8.55
20nodes_80regions_10000reads          7.76
20nodes_80regions_30000reads          8.19
30nodes_120regions_100000reads        8.23
30nodes_120regions_10000reads         5.57
30nodes_120regions_30000reads         8.17
30nodes_30regions_100000reads         8.91
30nodes_30regions_10000reads          8.68
30nodes_30r

In [10]:
scores_df.mean()

10nodes_10regions_100000reads     7.37
10nodes_10regions_10000reads      7.26
10nodes_10regions_30000reads      7.48
10nodes_20regions_100000reads     7.80
10nodes_20regions_10000reads      7.85
10nodes_20regions_30000reads      7.94
10nodes_40regions_100000reads     4.62
10nodes_40regions_10000reads      4.85
10nodes_40regions_30000reads      4.64
20nodes_20regions_100000reads     8.89
20nodes_20regions_10000reads      8.90
20nodes_20regions_30000reads      8.90
20nodes_40regions_100000reads     8.13
20nodes_40regions_10000reads      8.07
20nodes_40regions_30000reads      8.14
20nodes_80regions_100000reads     3.92
20nodes_80regions_10000reads      4.37
20nodes_80regions_30000reads      4.07
30nodes_120regions_100000reads    2.98
30nodes_120regions_10000reads     2.69
30nodes_120regions_30000reads     3.51
30nodes_30regions_100000reads     8.91
30nodes_30regions_10000reads      8.68
30nodes_30regions_30000reads      8.92
30nodes_60regions_100000reads     8.09
30nodes_60regions_10000re

### Results after 512*n^2 iterations

The robustness score (the column on the right) less than 5 will be repeated this time with twice as much iterations.

`
10nodes_10regions_100000reads     7.37

10nodes_10regions_10000reads      7.26

10nodes_10regions_30000reads      7.48

10nodes_20regions_100000reads     7.80

10nodes_20regions_10000reads      7.85

10nodes_20regions_30000reads      7.94

10nodes_40regions_100000reads     4.62

10nodes_40regions_10000reads      4.85

10nodes_40regions_30000reads      4.64

20nodes_20regions_100000reads     8.89

20nodes_20regions_10000reads      8.90

20nodes_20regions_30000reads      8.90

20nodes_40regions_100000reads     8.13

20nodes_40regions_10000reads      8.07

20nodes_40regions_30000reads      8.14

20nodes_80regions_100000reads     3.92

20nodes_80regions_10000reads      4.37

20nodes_80regions_30000reads      4.07

30nodes_120regions_100000reads    2.98

30nodes_120regions_10000reads     2.69

30nodes_120regions_30000reads     3.51

30nodes_30regions_100000reads     8.91

30nodes_30regions_10000reads      8.68

30nodes_30regions_30000reads      8.92

30nodes_60regions_100000reads     8.09

30nodes_60regions_10000reads      7.71

30nodes_60regions_30000reads      8.09

dtype: float64 `